SampleBufferContentKeySessionSupportEnabled is not attached to DeprecatedGlobalSettings

brentfulgham opened the pull request:
#### 14816ea69ff90349214ca5f7371d4a21f9244bfc
<pre>
SampleBufferContentKeySessionSupportEnabled is not attached to DeprecatedGlobalSettings
<a href="https://bugs.webkit.org/show_bug.cgi?id=250337">https://bugs.webkit.org/show_bug.cgi?id=250337</a>
&lt;rdar://problem/104044710&gt;

Reviewed by Eric Carlson.

Correct a missing &apos;webcorebindings&apos; field in the new generator file.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml: Include
missing webcoreOnChange binding.
* Source/WebCore/page/DeprecatedGlobalSettings.cpp: Remove unused method
* Source/WebCore/page/DeprecatedGlobalSettings.h: Remove unused method and
member variable.
* Source/WebCore/page/SettingsBase.cpp: Add handler for KeySessionSupport.
* Source/WebCore/page/SettingsBase.h

Canonical link: <a href="https://commits.webkit.org/258734@main">https://commits.webkit.org/258734@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9d40271d6ddd28411328304b3f079cf97460f632

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102706 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11829 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35735 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111965 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/172206 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106673 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12839 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2735 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94965 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109655 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108482 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9849 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93063 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37509 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/91701 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24575 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/79252 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/92926 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5286 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25993 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/89271 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/2970 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5436 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2440 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/29649 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11456 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45484 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/92195 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6000 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7177 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/20651 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/page/DeprecatedGlobalSettings.h:
```diff
@@ -170,7 +170,7 @@ class DeprecatedGlobalSettings {
 
 #if HAVE(AVCONTENTKEYSPECIFIER)
-    static void setSampleBufferContentKeySessionSupportEnabled(bool);
+    WEBCORE_EXPORT static void setSampleBufferContentKeySessionSupportEnabled(bool);
```
Is there a reason to restore this? Adding `webcorebinding: DeprecatedGlobalSettings` doesn't require `WEBCORE_EXPORT`

brentfulgham replied:
Yes it does -- see the build failure from the previous version of this patch.

nt1m replied:
Ah https://searchfox.org/wubkat/rev/a19946e70e2fc15526742690123bc3f4afc4ff0f/Source/WebKit/GPUProcess/GPUProcessPreferences.cpp#75 is what uses it outside of WebCore.

brentfulgham replied:
```
ld: warning: building for iOS, but linking in .tbd file (/Volumes/Data/worker/iOS-16-Build-EWS/build/WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/GraphicsServices.framework/GraphicsServices.tbd) built for iOS Simulator
Undefined symbols for architecture arm64:
  "WebCore::DeprecatedGlobalSettings::setSampleBufferContentKeySessionSupportEnabled(bool)", referenced from:
      -[WebView(WebViewInternalPreferencesChangedGenerated) _preferencesChangedGenerated:] in WebViewPreferencesChangedGenerated.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I think if you remove WEBCORE_EXPORT from anything actually used by the WebKit or WebKitLegacy layers you will get this build error.

nt1m commented on Source/WebCore/page/SettingsBase.cpp:
```diff
@@ -496,4 +496,12 @@ void SettingsBase::resourceUsageOverlayVisibleChanged()
+void SettingsBase::sampleBufferContentKeySessionSupportEnabledChanged()
+{
+    if (m_page)
+        MediaSessionManagerCocoa::setSampleBufferContentKeySessionSupportEnabled(m_page->settings().sampleBufferContentKeySessionSupportEnabled());
```
I guess EWS will tell, but does this need an include for `MediaSessionManagerCocoa.h` ?

brentfulgham replied:
Oh, duh. Yes.

nt1m reviewed (commented):
Thanks! Great to see more things out of DeprecatedGlobalSettings

> * Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml: Include
missing DeprecatedGlobalSetting binding.

That's no longer accurate.

nt1m reviewed (commented):
This looks good to me, though I'd prefer someone who works on the media code to look at it.

eric-carlson reviewed (approved):
r=me

brentfulgham committed:
SampleBufferContentKeySessionSupportEnabled is not attached to DeprecatedGlobalSettings
https://bugs.webkit.org/show_bug.cgi?id=250337
<rdar://problem/104044710>

Reviewed by Eric Carlson.

Correct a missing 'webcorebindings' field in the new generator file.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml: Include
missing webcoreOnChange binding.
* Source/WebCore/page/DeprecatedGlobalSettings.cpp: Remove unused method
* Source/WebCore/page/DeprecatedGlobalSettings.h: Remove unused method and
member variable.
* Source/WebCore/page/SettingsBase.cpp: Add handler for KeySessionSupport.
* Source/WebCore/page/SettingsBase.h

Canonical link: https://commits.webkit.org/258734@main
